packages/main/src/Menu.ts

```diff
@@ -338,8 +389,15 @@ class Menu extends UI5Element {
 		}, MENU_OPEN_DELAY);
 	}
 
+	_isGroupParent(parent: MenuItem | Menu | MenuItemGroup): boolean {
+		return parent && "isGroup" in parent && parent.isGroup;
+	}
+
 	_itemClick(e: CustomEvent<ListItemClickEventDetail>) {
 		const item = e.detail.item as MenuItem;
+		const prevSelected = item.isSelected;
+
+		item.isSelected = !prevSelected;
```

Comment:
I think the MenuItem needs a new property "selected", you will also benefit from it in the styles as you will be able to use [selected] selector and not adding some additional classes

Reply:
**MenuItem** extends **ListItem** which already have `selected` property, but `selected` in **ListItem** has totally different behavior, so I can neither use `selected` nor reuse related CSS. So I definitely need another "selected" state property here.